submission/distributor_test.go

```diff
@@ -354,23 +382,23 @@ func TestDistributorAddPreChain(t *testing.T) {
 			ctx, cancel := context.WithTimeout(context.Background(), time.Second)
 			defer cancel()
 
-			dist.Run(ctx)
+			if errs := dist.RefreshRoots(ctx); len(errs) != 1 || errs["ct.googleapis.com/icarus/"] == nil {
+				// 1 error is expected, because the Icarus log has an invalid root (see RootCerts).
+				t.Fatalf("dist.RefreshRoots() = %v, want 1 error for 'ct.googleapis.com/icarus/'", errs)
+			}
 
 			scts, err := dist.AddPreChain(context.Background(), tc.rawChain)
 			if gotErr := (err != nil); gotErr != tc.wantErr {
-				t.Errorf("Expected to get errors is %v while actually getting errors is %v", tc.wantErr, gotErr)
+				t.Fatalf("dist.AddPreChain(%q) = (_, %v), want err? %t", tc.rawChain, err, tc.wantErr)
+			} else if gotErr {
+				return
 			}
 
 			if got, want := len(scts), len(tc.scts); got != want {
-				t.Errorf("Expected to get %d SCTs on AddPreChain request, got %d", want, got)
+				t.Errorf("dist.AddPreChain(%q) = %d SCTs, want %d SCTs", tc.rawChain, got, want)
 			}
-			gotMap := getSCTMap(tc.scts)
-			for _, asct := range scts {
-				if wantedSCT, ok := gotMap[asct.LogURL]; !ok {
-					t.Errorf("dist.AddPreChain() = (_, %v), want err? %t", err, tc.wantErr)
-				} else if diff := cmp.Diff(asct, wantedSCT); diff != "" {
-					t.Errorf("Got unexpected SCT for Log %q", asct.LogURL)
-				}
+			if diff := cmp.Diff(scts, tc.scts); diff != "" {
```

Comment:
It looks you expect scts are always somehow sorted.

Reply:
The test that expects SCTs to be returned only expects 1, so ordering isn't currently important. However, it'd be a good idea to add a test for more than 1 SCT, so I've modified this to sort the SCTs before comparing them.